lib/Wikia/src/Service/Swagger/ApiClient.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Wikia\Service\Swagger;
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Configuration;
+use Wikia\Logger\Loggable;
+use Wikia\Util\Statistics\BernoulliTrial;
+
+class ApiClient extends \Swagger\Client\ApiClient {
+
+	use Loggable;
+
+	/** @var BernoulliTrial */
+	private $logSampler;
+
+	/** @var string */
+	private $serviceName;
+
+	public function __construct(Configuration $config, BernoulliTrial $logSampler, $serviceName) {
+		parent::__construct($config);
+		$this->logSampler = $logSampler;
+		$this->serviceName = $serviceName;
+	}
+
+	public function callApi($resourcePath, $method, $queryParams, $postData, $headerParams, $responseType=null) {
+		$start = microtime(true);
+		$response = $exception = null;
+		$code = 200;
+
+		try {
+			$response = parent::callApi($resourcePath, $method, $queryParams, $postData, $headerParams, $responseType);
```

Comment:
`$response` is an `array` of the json-decoded response and the headers on success (anything >= 200 && <= 299). While some of our successes are `204`, it's probably ok to log here as `200` to indicate a success. we *could* write some logic to turn a `200 OK` or a `204 No-Content` string into the appropriate `int`, but I didn't think it mattered too much. do you think we should do that?